lib/sql_parser.mly

```diff
@@ -369,7 +369,11 @@ expr:
     | e1=expr mnot(IN) l=sequence(expr) { poly Bool (e1::l) }
     | e1=expr mnot(IN) LPAREN select=select_stmt RPAREN { poly Bool [e1; SelectExpr (select, `AsValue)] }
     | e1=expr IN table=table_name { Tables.check table; e1 }
-    | e1=expr mnot(IN) p=PARAM { poly Bool [ e1; Inparam (new_param p Any) ] }
+    | lp=LPAREN e1=expr b=in_or_not_in p=PARAM rp=RPAREN
```

Comment:
so the parentheses requirement comes only from the need to infer locations? we can use http://gallium.inria.fr/~fpottier/menhir/manual.html#sec52 then

Reply:
figured it out, required some care to preserve the lexeme start in the lexer